### ISSUE TITLE
Inventory Applying triggers IndexOutOfRange

### ISSUE THREAD

**Issue by MikeMatrix** [Joshuarox100/VRC-Inventory-Inventor#24]
Currently don't have the time to debug myself, so I'll just drop the error here.
I might take a look at it on the weekend, if it isn't fixed by then.

From an initial look at it, it seems to be triggered in the routine that reassigns menu references for persistence.
`// Reassign created menus to each other as submenus so the reference persists post restart.`

```
System.ArgumentOutOfRangeException: Index was out of range. Must be non-negative and less than the size of the collection.
Parameter name: index
  at System.ThrowHelper.ThrowArgumentOutOfRangeException (System.ExceptionArgument argument, System.ExceptionResource resource) [0x00029] in <eae584ce26bc40229c1b1aa476bfa589>:0 
  at System.ThrowHelper.ThrowArgumentOutOfRangeException () [0x00000] in <eae584ce26bc40229c1b1aa476bfa589>:0 
  at System.Collections.Generic.List`1[T].get_Item (System.Int32 index) [0x00009] in <eae584ce26bc40229c1b1aa476bfa589>:0 
  at InventoryInventor.Manager.CreateMenus (VRC.SDK3.Avatars.ScriptableObjects.VRCExpressionsMenu& mainMenu, System.Int32 totalItems) [0x00a98] in E:\UnityProjects\Saki_Base_2019\Assets\Inventory Inventor\Editor\Scripts\InventoryInventor.cs:925 
  at InventoryInventor.Manager.CreateInventory () [0x0179e] in E:\UnityProjects\Saki_Base_2019\Assets\Inventory Inventor\Editor\Scripts\InventoryInventor.cs:643 
UnityEngine.Debug:LogError (object)
InventoryInventor.Manager:CreateInventory () (at Assets/Inventory Inventor/Editor/Scripts/InventoryInventor.cs:706)
InventoryInventor.InventoryInventorWindow:DrawCreateWindow () (at Assets/Inventory Inventor/Editor/Windows/InventoryInventorWindow.cs:192)
InventoryInventor.InventoryInventorWindow:OnGUI () (at Assets/Inventory Inventor/Editor/Windows/InventoryInventorWindow.cs:83)
UnityEngine.GUIUtility:ProcessEvent (int,intptr)
```

**Comment by Joshuarox100** [comment]
After looking into it a bit, I can't seem to reproduce your error. Additionally, without more information or my hands on the preset your using, I can't really debug the issue much further.

The only two scenarios that this case could be thrown, both of which shouldn't ever occur to begin with, are if the menu created has more controls than the number of items in the page it's made from or if the menus are created in a different order than they are listed in the Preset.

If you would be able to figure out which of these is the case for you and potentially why it is happening, it would greatly help.

**Comment by MikeMatrix** [comment]
That's what threw me off as well. Didn't look further into it because of that.
I'll probably take a look and throw in some debug code over the weekend.

**Comment by MikeMatrix** [comment]
Ok from what I can gather so far, it's mostly about things getting screwed up by the creation referencing names with similar names.

The 2 big issues (which kind of relate to the same initial problem) are

- The call to `FindAssets` is getting done with an unquoted name, which can result in multiple results, which messes with the referencing of files in the script (ex: `Outfits - Page 1` and `Outfits - Page 2` would result in `Outfits - Page 1 t:VRCExpressionMenu` and `Outfits - Page 2 t:VRCExpressionMenu` which would both result in the same result set, so in my case, referencing page 1 twice, instead of upholding the referencing)
- The call to `DeleteAsset` apparently is having a similar issue, although I'm not 100% sure, why that would be, since it should be doing it by file path? From my observations though, it will on some cases delete a previously created similar asset, that we had in the set, causing our `pages` to now contain null references.

Honestly, I think apart from the query needing to be quoted regardless, I think solving this generally by normalizing the names related to stuff, is the better approach here.

I'll probably work on that tomorrow and try and get a PR up for this.

**Comment by Joshuarox100** [comment]
Main issue with FindAssets has been fixed in v1.3.1, which seems to have also resolved the DeleteAsset issue according to the author. Name normalization is a work in progress.
